src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,72 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
 public class WorkWithFile {
+    private static final String COMMA = ",";
+    private static final int OPERATION_TYPE_INDEX = 0;
+    private static final int AMOUNT_INDEX = 1;
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
 
+        int[] totals = readFromFile(fromFileName);
+        String report = generateReport(totals[0], totals[1]);
+        writeToFile(toFileName, report);
+    }
+
+    private static int[] readFromFile(String fromFileName) {
+
+        int supplyTotal = 0;
+        int buyTotal = 0;
+
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                String[] parts = line.split(COMMA);
+                String operationType = parts[OPERATION_TYPE_INDEX];
+                int amount = Integer.parseInt(parts[AMOUNT_INDEX]);
+
+                if (operationType.equals(SUPPLY)) {
+                    supplyTotal += amount;
+                } else if (operationType.equals(BUY)) {
+                    buyTotal += amount;
+                }
```

Comment:
why do you have a calculation logic here? Isn't this method supposed only to read data from a file?